app-backend/db/src/test/scala/com/azavea/rf/database/AnnotationLabelDaoSpec.scala

```diff
@@ -499,4 +500,76 @@ class AnnotationLabelDaoSpec
       )
     }
   }
+
+  test("toggle label by session") {
+    check {
+      forAll(
+        (
+            userCreate: User.Create,
+            annotationProjectCreate: AnnotationProject.Create,
+            annotationCreates: List[AnnotationLabelWithClasses.Create],
+            taskFeatureCollectionCreate: Task.TaskFeatureCollectionCreate,
+            taskSessionCreate: TaskSession.Create
+        ) => {
+
+          val toInsert = annotationProjectCreate.copy(
+            labelClassGroups = annotationProjectCreate.labelClassGroups.take(1)
+          )
+
+          val insertIO = for {
+            user <- UserDao.create(userCreate)
+            annotationProject <-
+              AnnotationProjectDao
+                .insert(toInsert, user)
+            classIds = annotationProject.labelClassGroups flatMap {
+              _.labelClasses
+            } map { _.id }
+            fixedUpTasks = fixupTaskFeaturesCollection(
+              taskFeatureCollectionCreate,
+              annotationProject,
+              None
+            )
+            task <- TaskDao.insertTasks(
+              fixedUpTasks.copy(features = fixedUpTasks.features.take(1)),
+              user
+            ) map { _.features.head }
+            dbTaskSession <-
+              TaskSessionDao
+                .insertTaskSession(
+                  taskSessionCreate,
+                  user,
+                  task.properties.status,
+                  task.id
+                )
+            _ <- AnnotationLabelDao.insertAnnotations(
+              annotationProject.id,
+              task.id,
+              annotationCreates map { create =>
+                addClasses(create, classIds)
+                  .copy(sessionId = Some(dbTaskSession.id))
+              },
+              user
+            )
+            _ <- AnnotationLabelDao.toggleBySessionIds(
+              NonEmptyList(dbTaskSession.id, Nil),
+              false
+            )
+            activeLabels <-
+              AnnotationLabelDao.listWithClassesByProjectIdAndTaskId(
+                annotationProject.id,
+                task.id
+              )
+          } yield (activeLabels)
```

Comment:
This could use a test going the other direction -- not just active labels after toggling off, but also active labels after toggling on, since both directions are part of the endpoint